contracts/Liquidator.sol

```diff
@@ -1,41 +1,82 @@
-pragma solidity ^0.5.16;
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity ^0.8.10;
+
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
 
-import "./ComptrollerInterface.sol";
-import "./VAIControllerInterface.sol";
-import "./VBNB.sol";
-import "./VBep20.sol";
 import "./Utils/ReentrancyGuard.sol";
 import "./Utils/WithAdmin.sol";
-import "./Utils/SafeMath.sol";
-import "./Utils/IBEP20.sol";
-import "./Utils/SafeBEP20.sol";
+
+interface IComptroller {
+    function liquidationIncentiveMantissa() external view returns (uint256);
+}
+
+interface IVToken is IERC20 {}
+
+interface IVBep20 is IVToken {
+    function underlying() external view returns (address);
+    function liquidateBorrow(address borrower, uint256 repayAmount, IVToken vTokenCollateral)
+        external
+        returns (uint256);
+}
+
+interface IVBNB is IVToken {
+    function liquidateBorrow(address borrower, IVToken vTokenCollateral) external payable;
+}
+
+interface IVAIController {
+    function liquidateVAI(address borrower, uint256 repayAmount, IVToken vTokenCollateral)
+        external
+        returns (uint256, uint256);
+    function getVAIAddress() external view returns (address);
+}
 
 contract Liquidator is WithAdmin, ReentrancyGuard {
 
     /// @notice Address of vBNB contract.
-    VBNB public vBnb;
+    IVBNB public vBnb;
 
     /// @notice Address of Venus Unitroller contract.
     IComptroller comptroller;
 
     /// @notice Address of VAIUnitroller contract.
-    VAIControllerInterface vaiController;
+    IVAIController vaiController;
 
     /// @notice Address of Venus Treasury.
     address public treasury;
 
     /// @notice Percent of seized amount that goes to treasury.
     uint256 public treasuryPercentMantissa;
 
+    mapping (address => mapping (address => bool)) public liquidationAllowed;
+    mapping (address => bool) public liquidationRestricted;
+
     /// @notice Emitted when once changes the percent of the seized amount
     ///         that goes to treasury.
     event NewLiquidationTreasuryPercent(uint256 oldPercent, uint256 newPercent);
 
     /// @notice Event emitted when a borrow is liquidated
-    event LiquidateBorrowedTokens(address liquidator, address borrower, uint256 repayAmount, address vTokenCollateral, uint256 seizeTokensForTreasury, uint256 seizeTokensForLiquidator);
+    event LiquidateBorrowedTokens(
+        address indexed liquidator,
+        address indexed borrower,
+        uint256 repayAmount,
+        address indexed vTokenCollateral,
+        uint256 seizeTokensForTreasury,
+        uint256 seizeTokensForLiquidator
+    );
```

Comment:
I think it's a good idea, but I'd like to avoid things like `LiquidateBorrowedTokensV2` 😒
@coreyar can you please confirm adding an extra parameter won't break anything in our indexing?